c2tc-mobile/screens/LiveLocation.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import {
+  StyleSheet,
+  View,
+  Dimensions,
+} from 'react-native';
+
+import MapView, { ProviderPropType } from 'react-native-maps';
+
+const { width, height } = Dimensions.get('window');
+const ASPECT_RATIO = width / height;
+const LATITUDE_DELTA = 0.0922;
+const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;
+
+class LiveLocation extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    
+    this.state = {
+        mapRegion: null,
+        lastLat: null,
+        lastLong: null,
+      }        
+  }
+
+  componentDidMount() {
+    this.watchID = navigator.geolocation.watchPosition((position) => {
+      let region = {
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude,
+        latitudeDelta: LATITUDE_DELTA,
+        longitudeDelta: LONGITUDE_DELTA
+      }
+      this.onRegionChange(region, region.latitude, region.longitude);
+    });
+  }
+
+  onRegionChange(region, lastLat, lastLong) {
+    this.setState({
+      mapRegion: region,
+      lastLat: lastLat || this.state.lastLat,
+      lastLong: lastLong || this.state.lastLong
+    });
+  }
+
+  componentWillUnmount() {
+    navigator.geolocation.clearWatch(this.watchID);
+  }
+
+  onMapPress(e) {
+    let region = {
+      latitude: e.nativeEvent.coordinate.latitude,
+      longitude: e.nativeEvent.coordinate.longitude,
+      latitudeDelta: LATITUDE_DELTA,
+      longitudeDelta: LONGITUDE_DELTA
+    }
+    this.onRegionChange(region, region.latitude, region.longitude);
+  }
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <MapView
+          style={styles.map}
+          region={this.state.mapRegion}
+          showsUserLocation={true}
+          followUserLocation={true}
+          onPress={this.onMapPress.bind(this)}>
```

Comment:
Instead of using this.onMapPress.bind(this) try to use anonymous functions. 
it would look something like this: 
     onMapPress = e => {//code}
     onPress={this.onMapPress}>